src/LetsEncrypt/Internal/CertificateSelector.cs

```diff
@@ -104,5 +106,25 @@ public bool TryGet(string domainName, out X509Certificate2? certificate)
         {
             return _certs.TryGetValue(domainName, out certificate);
         }
+
+        private void PreloadIntermediateCertificates(X509Certificate2 certificate)
+        {
+            using var chain = new X509Chain
+            {
+                ChainPolicy =
+                {
+                    RevocationMode = X509RevocationMode.NoCheck
+                }
+            };
+
+            if (chain.Build(certificate))
```

Comment:
Note that it's very common for `chain.Build` to throw, rather than return bool especially in environments where the system time has slipped out of sync with internet time (because you can't build certs issued in the future). I assume there's something further up that would catch this though.

Reply:
If this throws, the cert would not be made available for use.

Reply:
FWIW: A cert from the future would just return false and report X509ChainStatusFlags.NotTimeValid in one of the "what went wrong" parts of the X509ChainElement structure.  The only exceptions from X509Chain.Build should be akin to OutOfMemoryException (or other "I asked a native method to do a thing and it told me it was unable to" reasons... like "uh, your arguments make no sense, do you have a misaligned pointer somewhere?")

Reply:
Thanks for confirming, @bartonjs.

Reply:
Thanks, yep false alarm. I was getting this mixed up with a different Certes library method that uses bouncycastle.

Reply:
Actually, this was a create call out @hwebprofusion-chrisc. The `.Build()` call throws consistently for me on macOS when using Let's Encrypt's staging server, which produces certs from an untrusted CA. I don't expect this will cause issues in production for most users, but I've added a fix in the next release: https://github.com/natemcmaster/LettuceEncrypt/commit/c927cd2a67f0a5c5ae4d9ad01ee1fe399d8b3f69

Reply:
Cool, I also use (ACME) BuyPass Go and smallstep, so encountering untrusted CAs is maybe what triggered me.

Reply:
Untrusted CAs shouldn't be triggering exceptions, but the chain builder on macOS is the most likely to throw (any time the underlying system gives us a code that we don't know which X509ChainStatusFlags value to map to).  3.1.5/5.0-preview6 should be less throwy in this regard.

If you see X509Chain.Build throw an exception, an issue at https://github.com/dotnet/runtime/issues, with a repro, is appreciated. If you want to wait for 3.1.5/5.0-preview6, no worries.